Sources/ApolloCodegenLib/Frontend/JavaScriptBridge.swift

```diff
@@ -0,0 +1,345 @@
+import JavaScriptCore
+
+// JavaScriptCore APIs haven't been annotated for nullability, but most of its methods will never return `null`
+// and can be safely force unwrapped. (Even when an exception is thrown they would still return
+// a `JSValue` representing a JavaScript `undefined` value.)
```

Comment:
Is it worth doing some handling of this type of thing? I think at the swift level we still want to know if a property is `nil`

Reply:
We could make all properties optional, but that would be a huge burden on using them. It is part of the contract that a `GraphQLNamedType` or a `Field` has a `name` for example, and that is enforced by our TypeScript types. I don't think making that optional is helpful. If it turns out it is `undefined` (or of the wrong type) at runtime, that means something has gone awfully wrong and we should abort and fix it. There isn't much else we can do, so error handling doesn't really make sense.

Reply:
(Some properties are truly optional of course, like `Field.alias`, so those can return `nil`.)

Reply:
OK - then it seems like if it's returning `undefined` that might be a place to throw an error? Or are you assuming that'll be caught later in the parsing?

Reply:
It may make sense to add an extra check to prevent the loose JavaScript conversion rules from applying here. That would fail immediately with a precondition message instead of returning `undefined`.